### ISSUE TITLE
Add Autocomplete control

### ISSUE THREAD

**Issue by chanan** [chanan/BlazorStrap.Extensions#1]

**Comment by Liero** [comment]
Let's call it typeahead, to keep compatibility with:

 https://ng-bootstrap.github.io/#/components/typeahead/examples
and
 https://github.com/ericgio/react-bootstrap-typeahead

I'm willing to contribute.

1. I wonder, why this is supposed to be part of extensions and not the core library?
2. A design question: would you like API similar to ng-bootstrap typeahread, where you can control debounce, minLength etc by IObservable<string> pipe. See: [example](https://ng-bootstrap.github.io/#/components/typeahead/examples#basic )

  It is a very elegant solution and it will make the component implementation much simpler, but it forces the consumer to reference System.Reactive package in order to write the pipe. This will add cca 57KB to the webassembly project (3.1.0-preview4.19579.2), although it should be less in stable release.

**Comment by chanan** [comment]
Hi @Liero 

Thanks for wanting to contribute! Originally, when John and I were talking we thought we should keep the main library to be just things that were in bootstrap main, and everything else should be outside of that. However, I think it might just better to put these types of controls in with everything else. Maybe another namespace?

I am not opposed to having the consumer reference System.Reactive, as long as they dont have to for "normal" usage and only need it for mre advanced scenrios. Is that possible?

**Comment by Liero** [comment]
> as long as they dont have to for "normal" usage and only need it for mre advanced scenrios. Is that possible?

No, I don't think it is possible. I actually realized that also this library would be dependend on System.Reactive :(
Too bad, I just love RX. Never mind, I will do without it
